scipy/stats/_axis_nan_policy.py

```diff
@@ -0,0 +1,347 @@
+# Many scipy.stats functions support `axis` and `nan_policy` parameters.
+# When the two are combined, it can be tricky to get all the behavior just
+# right. This file contains utility functions useful for scipy.stats functions
+# that support `axis` and `nan_policy`, including a decorator that
+# automatically adds `axis` and `nan_policy` arguments to a function.
+
+import numpy as np
+import scipy.stats
+import scipy.stats.stats
+from functools import wraps
+from scipy._lib._docscrape import FunctionDoc, Parameter
+import inspect
+
+
+def _broadcast_array_shapes_remove_axis(arrays, axis=None):
```

Comment:
These first three functions were in `stats.py` before, but this new file is the perfect place for them.